components/butterfly-test/src/lib.rs

```diff
@@ -250,24 +249,22 @@ impl SwimNet {
         self.gossip_rounds().iter().map(|r| r + count).collect()
     }
 
-    pub fn check_rounds(&self, rounds_in: &Vec<isize>) -> bool {
-        let mut finished = Vec::with_capacity(rounds_in.len());
-        for (i, round) in rounds_in.into_iter().enumerate() {
-            if self.members[i].paused() {
-                finished.push(true);
-            } else {
-                if self.members[i].swim_rounds() > *round {
-                    finished.push(true);
-                } else {
-                    finished.push(false);
-                }
+    fn check_rounds_impl(
+        &self,
+        rounds_in: &[isize],
+        get_rounds: impl Fn(&Server) -> isize,
+    ) -> bool {
+        for (member, round) in self.members.iter().zip(rounds_in) {
+            if !member.paused() && (get_rounds)(member) <= *round {
```

Comment:
Why is `get_rounds` parenthesized here?

Reply:
because it's a function pointer. It may not be strictly necessary to add parens in this case, but I know rust sometimes requires it. I'll see if they can be removed and if so make that change later.

Reply:
Yeah, I guess it's only when you're accessing a closure as a field of some other member. I'll get rid of those.

Reply:
Fixed in https://github.com/habitat-sh/habitat/pull/6068